api/catalog/api/serializers/media_serializers.py

```diff
@@ -73,6 +86,61 @@ class TagSerializer(serializers.Serializer):
     )
 
 
+def get_search_request_source_serializer(media_type):
+    class MediaSearchRequestSourceSerializer(serializers.Serializer):
+        """
+        This serializer parses and validates the source/not_source fields from the query
+        parameters.
+        """
+
+        field_names = [
+            "source",
+            "excluded_source",
+        ]
+        """
+        Keep the fields names in sync with the actual fields below as this list is
+        used to generate Swagger documentation.
+        """
+
+        _field_attrs = {
+            "help_text": (
+                "A comma separated list of data sources to search. "
+                f"Valid inputs: {format_enums(get_sources(media_type).keys())}"
+            ),
+            "required": False,
+        }
+
+        source = serializers.CharField(label="provider", **_field_attrs)
+        excluded_source = serializers.CharField(
+            label="excluded_provider", **_field_attrs
+        )
+
+        @staticmethod
+        def validate_source_field(input_sources):
+            allowed_sources = list(get_sources("image").keys())
```

Comment:
It seems like `get_sources(media_type)` call could be moved outside of the class definition instead of having to call it multiple times.

```suggestion
            allowed_sources = list(get_sources(media_type).keys())
```